frontend/e2e/helpers.cafe.ts

```diff
@@ -1,275 +1,375 @@
-    await t.click(selector);
-};
+  await click('#features-link')
+  await waitForElementVisible('#show-create-feature-btn')
+}
+
+export const click = async (selector: string) => {
+  await waitForElementVisible(selector)
+  await t
+    .expect(Selector(selector).hasAttribute('disabled'))
+    .notOk('ready for testing', { timeout: 5000 })
+    .hover(selector)
+    .click(selector)
+}
+
 export const gotoSegments = async () => {
-    await click('#segments-link');
-};
+  await click('#segments-link')
+}
 
-export const getLogger = () => RequestLogger(/api\/v1/, {
+export const getLogger = () =>
+  RequestLogger(/api\/v1/, {
+    logRequestBody: true,
+    logRequestHeaders: true,
     logResponseBody: true,
+    logResponseHeaders: true,
     stringifyRequestBody: true,
     stringifyResponseBody: true,
-    logResponseHeaders: true,
-    logRequestBody: true,
-    logRequestHeaders: true
-});
+  })
 
 export const gotoTraits = async () => {
-    await click('#users-link');
-    await click(byId('user-item-0'));
-    await waitForElementVisible('#add-trait');
-};
-
-export const createTrait = async (index:number, id: string, value:string|boolean|number) => {
-    await click('#add-trait');
-    await waitForElementVisible('#create-trait-modal');
-    await setText('[name="traitID"]', id);
-    await setText('[name="traitValue"]', `${value}`);
-    await click('#create-trait-btn');
-    await t.wait(2000);
-    await t.eval(() => location.reload());
-    await waitForElementVisible(byId(`user-trait-value-${index}`));
-    const expectedValue = typeof value === 'string' ? `"${value}"` : `${value}`;
-    await assertTextContent(byId(`user-trait-value-${index}`), expectedValue);
-};
-
-export const deleteTrait = async (index:number) => {
-    await click(byId(`delete-user-trait-${index}`));
-    await click('#confirm-btn-yes');
-    await waitForElementNotExist(byId(`user-trait-${index}`));
-};
+  await click('#users-link')
+  await click(byId('user-item-0'))
+  await waitForElementVisible('#add-trait')
+}
+
+export const createTrait = async (
+  index: number,
+  id: string,
+  value: string | boolean | number,
+) => {
+  await click('#add-trait')
+  await waitForElementVisible('#create-trait-modal')
+  await setText('[name="traitID"]', id)
+  await setText('[name="traitValue"]', `${value}`)
+  await click('#create-trait-btn')
+  await t.wait(2000)
+  await t.eval(() => location.reload())
+  await waitForElementVisible(byId(`user-trait-value-${index}`))
+  const expectedValue = typeof value === 'string' ? `"${value}"` : `${value}`
+  await assertTextContent(byId(`user-trait-value-${index}`), expectedValue)
+}
+
+export const deleteTrait = async (index: number) => {
+  await click(byId(`delete-user-trait-${index}`))
+  await click('#confirm-btn-yes')
+  await waitForElementNotExist(byId(`user-trait-${index}`))
+}
+
+const lastTestGroup = {}
+let lastTestName = undefined
```

Comment:
This feels dangerous, defining global variables accessed by multiple tests? Especially as we're trying to improve the parallelisation.

Reply:
That precisely allows managing the logs in an environment where several tests are being executed in parallel.